src/providers/wfs/qgsbackgroundcachedfeatureiterator.cpp

```diff
@@ -945,6 +946,8 @@ void QgsBackgroundCachedFeatureIterator::copyFeature( const QgsFeature &srcFeatu
         dstFeature.setAttribute( i, v );
       else if ( fieldType == QVariant::DateTime && !QgsVariantUtils::isNull( v ) )
         dstFeature.setAttribute( i, QVariant( QDateTime::fromMSecsSinceEpoch( v.toLongLong() ) ) );
+      else if ( fieldType == QVariant::Map && !QgsVariantUtils::isNull( v ) )
+        dstFeature.setAttribute( i, QVariant( QJsonDocument::fromJson( v.toString().toUtf8() ).toVariant() ) );
```

Comment:
```suggestion
        dstFeature.setAttribute( i, QJsonDocument::fromJson( v.toString().toUtf8() ).toVariant() );
```

No need for this constructor I think?

Reply:
indeed, fixed